src/tests/ftest/util/test_env_utils.py

```diff
@@ -0,0 +1,623 @@
+
+    if interface_speeds:
+        log.debug("Active network interface speeds on %s:", hosts)
+
+    for interface, speed in interface_speeds.items():
+        log.debug("  - %-8s (speed: %6s)", interface, speed)
+        # Only include the first active interface for each speed - first is
+        # determined by an alphabetic sort: ib0 will be checked before ib1
+        if speed is not None and speed not in available_interfaces:
+            available_interfaces[speed] = interface
+
+    log.debug("Available interfaces on %s: %s", hosts, available_interfaces)
+    return available_interfaces
+
+
+class TestEnvironment():
+    """Collection of test environment variables."""
+
+    def __init__(self):
+        """Initialize a TestEnvironment object.
+
+        This will set any unset test environment variables with default values.  Unless already set
+        externally the interface and provider values will not be set.  Use the set_with_hosts()
+        method to define default values for these test environment variables with host information.
+
+        When tests are run through launch.py it will have called set_with_hosts() and a subsequent
+        invocation of this class will read the values set via launch.py.
+        """
+        self.__env_map = OrderedDict()
+        self.__env_map['log_dir'] = {
+            'name': 'DAOS_TEST_LOG_DIR',
+            'method': self.__default_log_dir
+        }
+        self.__env_map['shared_dir'] = {
+            'name': 'DAOS_TEST_SHARED_DIR',
+            'method': self.__default_shared_dir
+        }
+        self.__env_map['app_dir'] = {
+            'name': 'DAOS_TEST_APP_DIR',
+            'method': self.__default_app_dir
+        }
+        self.__env_map['user_dir'] = {
+            'name': 'DAOS_TEST_USER_DIR',
+            'method': self.__default_user_dir
+        }
+        self.__env_map['interface'] = {
+            'name': 'DAOS_TEST_FABRIC_IFACE',
+            'method': self.__default_interface,
+            'kwargs': {'hosts': None}
+        }
+        self.__env_map['provider'] = {
+            'name': 'CRT_PHY_ADDR_STR',
+            'method': self.__default_provider,
+            'kwargs': {'hosts': None}
+        }
+        self.__env_map['insecure_mode'] = {
+            'name': 'DAOS_TEST_INSECURE_MODE',
+            'method': self.__default_insecure_mode
+        }
+        self.__env_map['bullseye_src'] = {
+            'name': 'DAOS_TEST_BULLSEYE_SRC',
+            'method': self.__default_bullseye_src
+        }
+        self.__env_map['bullseye_file'] = {
+            'name': 'COVFILE',
+            'method': self.__default_bullseye_file
+        }
+        self.__set_environment()
+
+    @property
+    def app_dir(self):
+        """Get the application directory path.
+
+        Returns:
+            str: the application directory path
+        """
+        return os.environ[self.__env_map['app_dir']['name']]
```

Comment:
I do wonder if this is too obfuscated. For example, looking at this definition, it's not clear how `'method': self.__default_app_dir` is used without looking at `__set_environment`. But it seems overkill for information that is static after initialization.